### DISCUSSION TITLE
🚘 Roadmap

### DISCUSSION THREAD

**Discussion by rokotyan** [f5/unovis#251] [General]
Welcome to the Unovis Roadmap, a list of our development goals to provide transparency for the community about our future endeavors. 

The roadmap covers several categories, including documentation updates, gallery refinements, introduction of new components and features, enhancements to existing tools, core improvements, and necessary bug fixes.

Latest update: 14 Aug 2023

---

### Docs
- XYLabels
- Contributing Guide https://github.com/f5/unovis/issues/255
- Events
- Dual Axis charts by overlaying two containers https://github.com/f5/unovis/pull/383

### Gallery
* Dedicated XYLabels example
* Patchy line example
* Graph example showcase multiple link labels and sticky tooltip
* Attribution example

### New Components
- Annotations, Configuring global text variables https://github.com/f5/unovis/pull/359
- ColorLegend
- TreeMap
- More color options

### New Features
- Chord Diagram: Transitions; Highlighting nodes via config https://github.com/f5/unovis/issues/257
- Accessibility
   - Automatic Patterns and Color Schemes (High Contrast)
   - Chart Summary autogeneration
   - Keyboard Navigation
- Leaflet Map: Links between points and clusters
- More built-in color options, shades
- LLM Assistant
- Vue Integration https://github.com/f5/unovis/pull/272
- Figma Plugin
- Server-side Rendering

### Enhancements
- Scatter: Labels Trimming
- Graph: Node label configurable width https://github.com/f5/unovis/discussions/111
- Graph: Add onDrag callbacks
- Opacity config for all XY components
- Axis: Supporting tick labels coloring through config (to allow different axis have different colors) 

### Core Enhancements
- All components: Update to the new CSS Variables definition syntax (use LeafletMap's `style.ts` as an example)
- Core: Add the second index `j` to accessors (useful for multi-accessor components)
- Callback types for events in Config Interfaces (that's going to be difficult, might require patching Emotion)
- XY Refactoring: Store Extent, Min / Max values, optimize bleed execution, maybe introduce a beforeRender function
- Update TypeScript to version 5
- Enable Strict Checks
- (Maybe) Supporting `Sizing.Extend` for Timeline

### Fixes
- Axis: When using `Scale.scaleTime()`, on the first render the `tickFormat` argument is `number` instead of `Date`
- Scatter: Labels get hidden sometimes for unknown reason

**Comment by 50rayn** [comment]
I suppose we can mark as done Vue integration
